echopype/utils/io.py

```diff
@@ -380,29 +382,30 @@ def env_indep_joinpath(*args: Tuple[str, ...]) -> str:
     return joined_path
 
 
-def validate_source_ds_da(
-    source_ds_da: Union[xr.Dataset, xr.DataArray, str, Path], storage_options: Optional[dict]
-) -> Tuple[Union[xr.Dataset, str, xr.DataArray], Optional[str]]:
+def validate_source_ds_da_ed(
+    source_ds_da_ed: Union[xr.Dataset, xr.DataArray, EchoData, str, Path],
```

Comment:
We can probably change `source_ds_da_ed` to just be `source`? Since the type is pretty clear here.